### PR TITLE
waffle: 1.6.1 -> 1.7.0

### PR DESCRIPTION
**Pull request by Flakebi**
###### Motivation for this change
1.7.0 was published: https://gitlab.freedesktop.org/mesa/waffle/-/releases#v1.7.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117346` at 30c15a7d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>piglit</li>
    <li>waffle</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/meson-cmake.md">meson-cmake</a>

Meson uses CMake as a fallback dependency resolution method and it likely is not necessary here. The message about cmake not being found is purely informational.

Near pkgs/development/libraries/waffle/default.nix:53:3:
```
   |
53 |   nativeBuildInputs = [
   |   ^
```
</li>
  </ul>
</details>

**Comment by Flakebi** [comment]
> Meson uses CMake as a fallback dependency resolution method and it likely is not necessary here. The message about cmake not being found is purely informational.

I think it’s needed in this case. 1.6.1 compiled fine without cmake but 1.7.0 fails to compile without it :)

File: pkgs/development/libraries/waffle/default.nix [hunk 1]
```diff
@@ -1,36 +1,39 @@
...
 , ninja
+, bash-completion
 , libGL
 , libglvnd ? null
 , makeWrapper
 , pkg-config
 , python3
 , x11Support ? true, libxcb ? null, libX11 ? null
-, waylandSupport ? true, wayland ? null
+, waylandSupport ? true, wayland ? null, wayland-protocols ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waffle/default.nix:14]:
```suggestion
, waylandSupport ? true, wayland, wayland-protocols
```

File: pkgs/development/libraries/waffle/default.nix [hunk 2]
```diff
@@ -1,36 +1,39 @@
...
+, cmake
 , meson
 , ninja
+, bash-completion
 , libGL
 , libglvnd ? null
 , makeWrapper
 , pkg-config
 , python3
 , x11Support ? true, libxcb ? null, libX11 ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waffle/default.nix:13]:
```suggestion
, x11Support ? true, libxcb, libX11
```

File: pkgs/development/libraries/waffle/default.nix [hunk 3]
```diff
@@ -1,36 +1,39 @@
...
-, waylandSupport ? true, wayland ? null
+, waylandSupport ? true, wayland ? null, wayland-protocols ? null
 , useGbm ? true, mesa ? null, libudev ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waffle/default.nix:15]:
```suggestion
, useGbm ? true, mesa, libudev
```

File: pkgs/development/libraries/waffle/default.nix [hunk 4]
```diff
@@ -1,36 +1,39 @@
...
 }:
 
 assert x11Support -> (libxcb != null && libX11 != null);
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waffle/default.nix:18]:
```suggestion
```

File: pkgs/development/libraries/waffle/default.nix [hunk 5]
```diff
@@ -1,36 +1,39 @@
...
-assert waylandSupport -> wayland != null;
+assert waylandSupport -> (wayland != null && wayland-protocols != null);
 assert useGbm -> (mesa != null && libudev != null);
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waffle/default.nix:20]:
```suggestion
```

File: pkgs/development/libraries/waffle/default.nix [hunk 6]
```diff
@@ -1,36 +1,39 @@
...
 assert with stdenv.hostPlatform; isUnix && !isDarwin -> libglvnd != null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waffle/default.nix:21]:
```suggestion
```

File: pkgs/development/libraries/waffle/default.nix [hunk 7]
```diff
@@ -1,36 +1,39 @@
 { stdenv
 , fetchFromGitLab
 , lib
+, cmake
 , meson
 , ninja
+, bash-completion
 , libGL
 , libglvnd ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/waffle/default.nix:9]:
```suggestion
, libglvnd
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Packages are never null and allowing them to be null and later enforcing they are not is not that useful.

**Comment by Flakebi** [comment]
Thanks, fixed

**Comment by Flakebi** [comment]
@SuperSandro2000, I fixed the

**Pull request merged**